### DISCUSSION TITLE
Users: Add option to use e-mail as username

### DISCUSSION THREAD

**Discussion by Hackwar** [joomla/joomla-cms#47016] [6. Duplicates / Archived / Not for discussion]
### Is your feature request related to a problem? Please describe.
People nowadays normally don't use usernames anymore, but their email address as username. Joomla should support that by removing the username field and using the email field for this.

### Describe the solution you'd like
Add a switch in the configuration of the user component to switch between usernames and using the email address as username and adopt login, registration and profile views and code accordingly.

**Comment by 4digitalworld** [comment]
finally this is added as a feature, i requested this almost 9 years ago [here](https://github.com/joomla/joomla-cms/issues/10109)!

**Comment by Hackwar** [comment]
I just made this proposal. This doesn't mean it is added.

**Comment by 4digitalworld** [comment]
> I just made this proposal. This doesn't mean it is added.

anyway, i think this is one of the most requested feature, it could just be an extra line in @Joomla settings like "**allow login via email**", many third party extensions already offer this, so now is the time to add it as a default option out of the box.

**Comment by Hackwar** [comment]
However it isn't just a one-line-change. Besides the option to switch between email and username, you have to add code to remove the username in forms everywhere, rename login fields, find solutions to stay backwards compatible, etc.

**Comment by 4digitalworld** [comment]
> email and username, you have to add code to remove the username in forms everywhere, rename login fields

instead of the `username` field it can be the `login or email` field

![Image](https://github.com/user-attachments/assets/eaf5035f-4c2d-4bf1-b0a3-0b421dcc9be9)
![Image](https://github.com/user-attachments/assets/31dc1d8c-fe19-44ff-89ca-4efe186f82a4)
![Image](https://github.com/user-attachments/assets/84fa8d14-864e-4a4d-8b15-5cd289fbb0f8)
![Image](https://github.com/user-attachments/assets/c911eb71-625f-4b8b-b678-75b86917c19f)

this is how Google and many other major websites did it.

**Comment by AasmitPadea** [comment]
I can work on this issue. My plan is to change the way of how the forms accept data so that we can get either of username or email and process it further as needed. I am eager to work on this issue please sir assign this issue to me i would love to add this feature

**Comment by Hackwar** [comment]
We don't assign issues to anybody. You can just work on something and send in a pull request and then we will discuss your code and maybe accept it.

**Comment by AasmitPadea** [comment]
Could you please help me i am having problems in understanding the codebase of joomla-cms. In which section of the file structure is all the logic to handle forms are written i will work accordingly

**Comment by AasmitPadea** [comment]
I have a quick question regarding the feature to use email as the username in Joomla. Will this change be applied only to the login and signup forms, or will it also affect other areas like user profiles and backend management?

Looking forward to your clarification.

**Comment by HLeithner** [comment]
I'm closing this and move it to duplicate/archive.

Older iteration of this feature request: https://github.com/joomla/joomla-cms/discussions/37016
